src/Parlot/Fluent/ParseContext.cs

```diff
@@ -32,7 +32,7 @@ public ParseContext(Scanner scanner, bool useNewLines = false)
         /// <summary>
         /// The parser that is used to parse whitespaces and comments.
         /// </summary>
-        public Parser<TextSpan> WhiteSpaceParser { get; set;}
+        public Parser<TextSpan, ParseContext> WhiteSpaceParser { get; set; }
```

Comment:
This looks like it could now be in a custom `ParseContext` that deals with `char`, since it doesn't make sense for byte based parsers. Then the `Literals` or `Terms` based parsers could use this new context type, to drive the other parsers generic type. There would then be a new set of binary parsers with another corresponding type.

Or maybe it's a `Scanner` concern too.

Reply:
I think you mixed this comment with my other PR. BTW, it is done as such in my other PR (see StringParseContext).